pilot/pkg/model/sidecar.go

```diff
@@ -248,6 +239,55 @@ func DefaultSidecarScopeForNamespace(ps *PushContext, configNamespace string) *S
 	return out
 }
 
+func (sc *SidecarScope) addService(s *Service) bool {
+	if s == nil {
+		return false
+	}
+	existing, found := sc.servicesByHostname[s.Hostname]
+	if !found {
+		sc.AddConfigDependencies(ConfigKey{
+			Kind:      gvk.ServiceEntry,
+			Name:      string(s.Hostname),
+			Namespace: s.Attributes.Namespace,
+		})
+		sc.services = append(sc.services, s)
+		sc.servicesByHostname[s.Hostname] = s
+		return true
+	}
+
+	// In some scenarios, there may be multiple Services defined for the same hostname due to ServiceEntry allowing
+	// arbitrary hostnames. In these cases,
+	// 1. if they are in same namespace, merge the ports.
+	// 2. otherwise, we want to pick the first Service, which is the oldest. This ensures
+	// newly created Services cannot take ownership unexpectedly.
+	// However, the Service is from Kubernetes it should take precedence over ones not. This prevents someone from
+	// "domain squatting" on the hostname before a Kubernetes Service is created.
+	// This relies on the assumption that
+	if existing.Attributes.Namespace == s.Attributes.Namespace && len(s.Ports) > 0 {
+		// merge the ports to service when each listener generates partial service
+		// we only merge if the found service is in the same namespace as the one we're trying to add
+		for _, p := range s.Ports {
+			found := false
+			for _, osp := range existing.Ports {
+				if p.Port == osp.Port {
+					found = true
+					break
+				}
+			}
+			if !found {
+				existing.Ports = append(existing.Ports, p)
```

Comment:
I believe this mutates the Service which makes this sticky even when the port is later removed.

Reply:
Note this was broken in `master` as well it appears. But still should fix it

Reply:
make sense